### PR TITLE
Allow for non-'static Module

### LINKED ISSUE #490: Remove `'static` requirement on `Module`
Implicitly through `AsAnyMut`

### PR DESCRIPTION
**Pull request by plafer**
Closes: #490

Supersedes: #493

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/applications/transfer/context.rs [hunk 1]
```diff
@@ -402,25 +402,11 @@ pub(crate) mod test {
...
-        msg: MsgTransfer<PrefixedCoin>,
-    ) -> Result<(), ChannelError> {
-        send_transfer(ctx, msg).map_err(|e: TokenTransferError| ChannelError::AppModule {
-            description: e.to_string(),
-        })
-    }
-
     fn get_defaults() -> (
         DummyTransferModule,
         Order,
```
Comment (Farhad-Shabani) [inline crates/ibc/src/applications/transfer/context.rs:412]: Now that Having `TokenTransferValidationContext` and `TokenTransferExecutionContext` implemented on `MockContext`, we can replace the context for tests here from the  `DummyTransferModule` to `MockContext`. Thus, would be ok if we remove the validation/execution context implementation of `DummyTransferModule` in `test_utils.rs` and maintain it purely as a Module object.

Comment (Farhad-Shabani) [inline crates/ibc/src/applications/transfer/context.rs]: Now that having `TokenTransferValidationContext` and `TokenTransferExecutionContext` implemented on `MockContext`, we can replace the context for tests here from the  `DummyTransferModule` to `MockContext`. Thus, would be ok if we remove the validation/execution context implementation of `DummyTransferModule` in `test_utils.rs` and maintain it purely as a Module object.

**Review by Farhad-Shabani — COMMENTED** [review]
Just a small tweak, otherwise it looks good :pray:

File: crates/ibc/src/applications/transfer/context.rs (hunk 1 above)
Comment (plafer) [reply crates/ibc/src/applications/transfer/context.rs]: [25db45c](https://github.com/cosmos/ibc-rs/pull/499/commits/25db45c6caf1f6037f7b214c5f900b76e26aac9b)

**Review by Farhad-Shabani — APPROVED** [review]
I left it to you to merge whenever you got sure of `basecoin-rs` update. ty 🙏

**Pull request merged**